quiche/src/lib.rs

```diff
@@ -3188,6 +3274,8 @@ impl<F: BufFactory> Connection<F> {
             drop_pkt_on_err(e, self.recv_count, self.is_server, &self.trace_id)
         })?;
 
+        // Why do we check if the pkt was not received AFTER decrypting it ??
```

Comment:
For security reason, we don't want to give the adversary a trivial decryption oracle. QUIC even makes these kinds of event silent as previous existing 'secure' protocols were vulnerable to various side-channels exploits combining timing information obtained from visible reaction-on-error to exploit cryptographic vulnerabilities. These vulnerabilities do not exist anymore, but the 'lets-be-silent" approach remained.